---
title: >
  Using DoReMiTra for analyzing and integrating publicly available transcriptomics radiation datasets
subtitle: >
  Supplementary file for 'DoReMiTra: An R/Bioconductor data package for orchestrating the analysis of radiation transcriptomic studies'
author:
- name: Ahmed Salah
  affiliation: 
  - Department of Radiation Oncology and Radiation Therapy, Mainz, Germany
  - Institute of Medical Biostatistics, Epidemiology and Informatics (IMBEI), Mainz, Germany
  email: ahhassan@uni-mainz.de
  orcid: 0000-0001-7003-4466
- name: Sebastian Zahnreich
  affiliation: 
  - Department of Radiation Oncology and Radiation Therapy, Mainz, Germany
  orcid: 0000-0001-8365-3258
- name: Federico Marini
  affiliation: 
  - Institute of Medical Biostatistics, Epidemiology and Informatics (IMBEI), Mainz, Germany
  email: marinif@uni-mainz.de
  orcid: 0000-0003-3252-7758
date: "`r format(Sys.Date(), '%d %B %Y')`"
format:
  html:
    toc: true
    number-sections: true
    embed-resources: true
    mainfont: "Atkinson Hyperlegible"
    link-external-newwindow: true
    theme: 
    - cosmo
    - ummz_theme.scss
bibliography: DRMTS.bib
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

# Loading the required packages

Here we load all packages required to reproduce the analysis in this notebook.

```{r loadlib, warning=FALSE, message=FALSE}
library("DoReMiTra")
library("mosdef")
library("org.Hs.eg.db")
library("DESeq2")
library("ggplot2")
library("ComplexHeatmap")
library("VennDiagram")
library("limma")
library("circlize")
library("GeneTonic")
library("clusterProfiler")
library("patchwork")
```

# List All Available Datasets

Users can use the **`list_DoReMiTra_datasets()`** function to have a concise overview of all datasets currently available in the DoReMiTra collection. It returns a structured data.frame including dataset identifiers and key descriptive information, allowing users to quickly inspect the available resources before retrieval.

```{r list-datasets}
datasets <- list_DoReMiTra_datasets()
knitr::kable(datasets)
```

# List medatada information

The **`list_DoReMiTra_metadata_fields()`** function provides an overview of the available sample-level metadata fields across all datasets included in DoReMiTra. For each metadata variable, the function reports the unique values observed across the compendium.

This utility facilitates intuitive dataset exploration and helps users identify appropriate filtering criteria before retrieving datasets with search_DoReMiTra_datasets() or get_DoReMiTra_dataset(). In particular, it allows users to quickly inspect supported categories such as radiation type, dose, time point, organism, tissue, and other study-specific annotations.

```{r explore-medatada-info}

list_DoReMiTra_metadata_fields()

```

# Searching for datasets

The user can use **`search_DoReMiTra_datasets()`** to filter the available DoReMiTra datasets based on metadata fields such as radiation type, organism, author, or experimental setting. This function helps narrow down datasets of interest before fetching them.

```{r search-datasets}
# X-ray datasets
search_DoReMiTra_datasets(radiation_type = "X-ray", 
                          organism = "Homo sapiens", 
                          exp_setting = "ExVivo")

# gamma ray datasets
search_DoReMiTra_datasets(radiation_type = "gamma ray", 
                          organism = "Homo sapiens", 
                          exp_setting = "ExVivo")
```


# Retrieving the datasets

After selecting the dataset of interest, **`get_DoReMiTra_data()`** can be used to fetch a `SummarizedExperiment` (SE) object with stable gene identifiers and consistently formatted sample-level metadata directly from ExperimentHub. 

```{r load-datasets}
# we start with a recent study by Salah et al., conducted on RNA-seq
dataset_name_1 <- "SE_Salah_2025_ExVivo"

# we would like also to compare it to another dataset from another radiation type to see shared and distinct gene signatures
dataset_name_2 <- "SE_Paul_2013_ExVivo_GSE44201_GPL6848"

se_salah <- get_DoReMiTra_data(dataset_name_1)
se_paul <- get_DoReMiTra_data(dataset_name_2, gene_symbol = TRUE)
```

These simple commands provide the user an analysis-ready object for the datasets generated by [@Salah2025] and by [@Paul2013].

# Obtaining summaries of the datasets

To quickly obtain a summary of an SE object, use **`summarize_DoReMiTra_se()`**, which provides a concise overview including the number of samples, genes, radiation types, tissue, etc.

```{r summarize}
summarize_DoReMiTra_se(se_salah)
summarize_DoReMiTra_se(se_paul)
```

Once we retrieved the analysis-ready datasets of intereset, we can explore them and analyse them in detail.

# Comparing the metadata of multiple datasets

To compare key metadata information from two or more SE objects from the DoReMiTra collection, **`compare_DoReMiTra_datasets()`** summarizes key metadata information such as radiation type, dose, and time point.

```{r compare}
se_list<- list(
  `Salah et al.` = se_salah, 
  `Paul et al.` = se_paul
)
compare_DoReMiTra_datasets(se_list = se_list) |> 
  kableExtra::kable()
```

# Using DoReMiTraExplorer for exploratory data analyses

One can also explore the datasets with `DoReMiTraExplorer`, a companion app package we have developed.

The command to install the app is exemplified in the following chunk:

```{r exploinstall, eval = FALSE}
# Install the DoReMiTra-explorer Shiny App
devtools::install_github("AhmedSAHassan/DoReMiTraExplorer")
```

The retrieved SummarizedExperiment objects can be explored with **`DoReMiTra_explorer()`**, an interactive Shiny application for visualization and exploratory analysis of DoReMiTra datasets. This app allows users to examine transcriptomic responses to radiation using:

-   Principal Component Analysis (PCA)
-   Boxplots for the distribution of expression values
-   Doseâ€“response gene plots
-   Heatmaps of highly variable genes

It provides flexibility for both high-level overviews and detailed inspection of radiation-induced transcriptional patterns.

To load the app, we can simply call

```{r runapp, warning=FALSE}
# load the DoReMiTraExplorer package
library("DoReMiTraExplorer")

# launch the app if in an interactive session
if(interactive()) {
  DoReMiTraExplorer::DoReMiTra_explorer(se = se_salah)
}
```

![Snapshot of the DoReMiTraExplorer app](images/clipboard-3009620933.png)
![Snapshot of the DoReMiTraExplorer app](images/GenePlot.png)
# Analyzing the datasets, in detail

Once we identified and retrieved the analysis-ready datasets of intereset, we can explore and analyse them.

## Working on the X-ray dataset

As we could observe from the PCA plot obtained via `DoReMiTraExplorer`, the donor and time effects are the main sources of variability. These factors should be accounted for in downstream statistical analyses. Specifically, we use in this case the DESeq2 statistical framework, and focus on the **2Gy vs 0Gy** comparison, starting from the subset of [@Salah2025] including samples 2 hours after irradiation.

This dataset consists of 3 donors, 5 doses (0, 0.5, 1, 2, and 4Gy), and 2 timepoints (2h and 6h)


```{r PCA-plot}
dds_salah <- DESeqDataSet(se_salah, design = ~1)

dds_salah$Dose <- factor(dds_salah$Dose)
dds_salah$Donors <- factor(dds_salah$Donors)
dds_salah$Dose <- relevel(dds_salah$Dose, ref = "0Gy")

dds_salah_2h <- dds_salah[, dds_salah$Time_point == "2hrs"]
dds_salah_6h <- dds_salah[, dds_salah$Time_point == "6hrs"]

vsd_salah <- vst(dds_salah)

pca_plot <- plotPCA(vsd_salah, 
                    intgroup = c("Donors", "Time_point"),
                    returnData = TRUE)

percentVar <- round(100 * attr(pca_plot, "percentVar"))

PCA <- ggplot(pca_plot, aes(x = PC1, y = PC2, color = Donors, shape = Time_point)) +
  geom_point(size = 3) +
  scale_shape_manual(values = c(16, 17)) + 
  xlab(paste0("PC1: ", percentVar[1], "% variance")) +
  ylab(paste0("PC2: ", percentVar[2], "% variance")) +
  theme_minimal()+
  theme(plot.background = element_rect(fill = "white", color = NA))

PCA

vsd_salah_6h <- vst(dds_salah_6h)

pca_donors <- plotPCA(vsd_salah_6h, 
        intgroup = "Donors",
        returnData = FALSE) + 
  theme_minimal() + 
  labs(title = "Salah et al., PCA by donors") 
pca_dose <- plotPCA(vsd_salah_6h, 
        intgroup = "Dose",
        returnData = FALSE) + 
  theme_minimal() + 
  labs(title = "Salah et al., PCA by dose")

pca_donors | pca_dose
```

We quickly inspect a heatmap of the highly variable genes - as we have strong donor effect, we will account for it (removing that via `removeBatchEffect()`) for better visualization. 
The uncorrected raw counts will be used for the subsequent steps.

```{r Heatmap}
vsd_corrected <- vsd_salah_6h
assay(vsd_corrected) <- limma::removeBatchEffect(assay(vsd_corrected), batch = vsd_corrected$Donors)
doses <- as.numeric(sub("Gy.*", "", colnames(vsd_corrected)))
col_order <- order(doses)
vsd_corrected <- vsd_corrected[, col_order]
colnames(vsd_corrected)

top_n <- 50 
rv <- rowVars(assay(vsd_corrected))
top_genes <- order(rv, decreasing = TRUE)[1:top_n]
mat <- assay(vsd_corrected)[top_genes, ]
mat_scaled <- t(scale(t(mat)))
annotation <- as.data.frame(colData(vsd_corrected)[, c("Sex","Dose"), drop = FALSE])
Heatmap(mat_scaled,
        name = "expression",
        cluster_columns = FALSE,
        top_annotation = HeatmapAnnotation(df = annotation))
```

The columns of the heatmap are left ordered by dose.

Afterwards, we proceed with the modeling and testing step with DESeq2.

```{r x-ray-analysis}
dds_salah_6h

design(dds_salah_6h)<- ~Donors + Dose
dds_salah_6h <- DESeq(dds_salah_6h)
resultsNames(dds_salah_6h)

# we will focus on one contrast here, 2Gy vs 0Gy
results_salah<- results(dds_salah_6h, 
                        name = "Dose_2Gy_vs_0Gy", 
                        alpha = 0.05)
summary(results_salah)
results_salah_shrink <- lfcShrink(dds_salah_6h, 
                                 res = results_salah, 
                                 coef = "Dose_2Gy_vs_0Gy")

result_salah_sig <- as.data.frame(results_salah_shrink)
result_salah_sig <- 
  result_salah_sig[!is.na(results_salah_shrink$padj) & result_salah_sig$padj <= 0.05, ]

# adding SYMBOL column
result_salah_sig$SYMBOL <- mapIds(
  org.Hs.eg.db,
  keys = rownames(result_salah_sig),
  column = "SYMBOL",
  keytype = "ENSEMBL",
  multiVals = "first"
)

result_salah_sig |> 
  head(10) |> 
  knitr::kable()
```

We can also visualize the up and down regulated genes via a volcano plot

```{r volcanoPlot}
mosdef::de_volcano(
  res_de = results_salah_shrink,
  mapping = "org.Hs.eg.db",
  labeled_genes = 50,
  logfc_cutoff = 0
)
```

As we can see here, among the highest X-ray sensitive genes upon 2Gy exposure we can find FDXR, CDKN1A, AEN, and CD83.

From this point, we can also perform a functional enrichment analysis to see what pathways are regulated

```{r xray-enrichment}
res_enrich_salah <- enrichGO(result_salah_sig$SYMBOL,
                             keyType = "SYMBOL",
                             ont = "BP",
                             OrgDb = "org.Hs.eg.db",
                             pvalueCutoff = 0.05,
                             pAdjustMethod = "BH",
                             qvalueCutoff = 0.2,
                             minGSSize = 10,
                             maxGSSize = 500
)

res_enrich_salah_sig <- res_enrich_salah@result[res_enrich_salah@result$p.adjust <= 0.05, ]
res_enrich_salah_sig |> 
  head(20) |> 
  knitr::kable()
```

Now we will also visualize the top 10 regulated pathways.

```{r xray-enrichment2}
# create GeneTonic object
anno_df <- mosdef::get_annotation_orgdb(
  de_container = dds_salah,
  orgdb_package = "org.Hs.eg.db",
  id_type = "ENSEMBL"
)

gtl_salah_6h <- GeneTonic::GeneTonicList(
  dds = dds_salah_6h,
  res_de = results_salah,
  res_enrich = shake_enrichResult(res_enrich_salah),
  annotation_obj = anno_df
)

# preparing the matrix for heatmap
scored_mat <- GeneTonic::gs_scores(se = vsd_corrected, gtl = gtl_salah_6h)

anno <- as.data.frame(colData(vsd_corrected)[, c("Sex", "Dose")])
dose_colors <- colorRampPalette(c("lightyellow", "gold", "orange", "red"))(5)
names(dose_colors)<- c("0Gy", "0.5Gy", "1Gy", "2Gy", "4Gy")
anno_colors <- list(
  Sex = c(Female = "pink", Male = "blue"),
  Dose = dose_colors
)
column_annotation <- HeatmapAnnotation(
  Sex = anno$Sex,
  Dose = anno$Dose,
  col = anno_colors, show_legend = TRUE
)

col_fun <- colorRamp2(
  c(-2, 0, 2),
  c("blue", "white", "red")
)

ht <- ComplexHeatmap::Heatmap(scored_mat[1:20,],
  name = "Expression",
  top_annotation = column_annotation,
  cluster_columns = FALSE, # Optional: Cluster columns
  cluster_rows = FALSE, # Optional: Cluster rows
  show_row_names = TRUE,
  show_column_names = TRUE, 
  row_dend_side = "left",
  row_names_side = "left",
  row_names_gp = gpar(fontsize = 5, fontface= "bold" ),
  width = unit(7, "cm"),
  heatmap_height = unit(12, "cm"),
  column_names_gp = gpar(fontsize = 7),
  column_names_rot = 90,
  column_title = "Donors", 
  column_title_side = "bottom",
  show_heatmap_legend = TRUE, 
  col = col_fun
)

draw(ht, heatmap_legend_side = "right", annotation_legend_side = "right")
```

TODO; We can see here that among the top 20 regulated pathways we can find processes such as DNA damage repair, immune response, cellular response to UV, and some other associated GO terms.


## Working on the gamma ray dataset (Paul et al. 2013)

We might also want to compare Xray signature to gamma ray signatures, to see if both ionizing radiation types have shared and distinct transcriptional responses we will therefore focus on the `se_paul` dataset, since it has the same time point of `se_salah` (6 hours after irradiation).

In this `se_paul` dataset, Age and Sex should be accounted for to capture signals only in response to radiation


```{r gamma-ray-analysis}
# for comparability, we will only focus on 6hr timepoint
se_paul_6h <- se_paul[, colData(se_paul)$Time_point == "6hr"]
# since it is a repeated measurements experiment, we will create a pseudo-donor factor to account for interdonor variability

expr_values <- assay(se_paul_6h)
# remove probes with NA values
expr_values <- expr_values[rowSums(is.na(expr_values)) == 0, ]

# remove probes with zero variance
expr_values <- expr_values[apply(expr_values, 1, sd) != 0, ]

# extracting metadata
meta_df <- as.data.frame(colData(se_paul_6h))

# number of donors
n_donors <- 5
# number of doses per donor
n_doses <- 5

# create pseudo-donor factor
meta_df$Donors <- factor(rep(1:n_donors, each = n_doses))

# check
table(meta_df$Donors)

# handle the covariates as factors
meta_df$Dose <- factor(meta_df$Dose)
meta_df$Dose <- relevel(meta_df$Dose, ref = "0Gy")
meta_df$Sex  <- factor(meta_df$Sex)
meta_df$Age <- as.numeric(gsub("[^0-9]", "", meta_df$Age))

# building the design matrix
design_paul <- model.matrix(~ Age + Sex + Dose, data = meta_df)
design_paul

# blocking for the donor effect
corfit_paul <- duplicateCorrelation(expr_values, design_paul, block = meta_df$Donors)

# fitting the model
fit_paul <- lmFit(expr_values, design_paul, block = meta_df$Patient, correlation = corfit_paul$consensus)

contrast.matrix_paul <- makeContrasts(Dose2Gy, levels = design_paul)
fit2_paul <- contrasts.fit(fit_paul, contrast.matrix_paul)
fit2_paul <- eBayes(fit2_paul)

# extracting the results
results_paul <- topTable(fit2_paul, number = Inf)
results_paul_sig <- results_paul[ results_paul$adj.P.Val<= 0.05,]

results_paul_sig |> 
  head(10) |> 
  knitr::kable()
```

Here, we will extract the gene symbol part without the probe id so we can compare the genes with the Salah_2025 dataset

```{r gamma-ray-analysis2}
results_paul_sig$SYMBOL <- sub("_.*", "", rownames(results_paul_sig))

# again, we can also check this effect at the pathway level
res_enrich_paul <- enrichGO(results_paul_sig$SYMBOL,
                            keyType = "SYMBOL",
                            ont = "BP",
                            OrgDb = "org.Hs.eg.db",
                            pvalueCutoff = 0.05,
                            pAdjustMethod = "BH",
                            qvalueCutoff = 0.2,
                            minGSSize = 10,
                            maxGSSize = 500
)

res_enrich_paul_sig <- res_enrich_paul@result[res_enrich_paul@result$p.adjust <= 0.05, ]

res_enrich_paul_sig |> 
  head(20) |> 
  knitr::kable()
```

Similarly to X-ray dataset from `se_salah`, many of the regulated pathways were involved in processes such as DNA damage repair, immune response, and response to UV.

# Integrating the results from both studies 

Checking the shared genes between both radiation types, at the same timepoint and radiation dose

```{r compare-datasets, message=FALSE}
result_salah_sig<- na.omit(result_salah_sig)

# creating a simple venn diagram first
venn.plot <- venn.diagram(
  x = list(
    Salah = result_salah_sig$SYMBOL,
    Paul = results_paul_sig$SYMBOL
  ),
  fill = c("#8da0cb", "orange"),
  alpha = 0.6,
  cat.cex = 1.2,
  cex = 1.5,
  main = "Overlap of DE genes - Shared and distinct signatures", 
  filename = NULL,  
  disable.logging = TRUE)

grid::grid.draw(venn.plot)

# shared signature between X and gamma rays
intersect(result_salah_sig$SYMBOL, results_paul_sig$SYMBOL)

# gamma-ray only signature
setdiff(results_paul_sig$SYMBOL, result_salah_sig$SYMBOL)

# X-ray only signature
setdiff(result_salah_sig$SYMBOL, results_paul_sig$SYMBOL)
```

Another way to display the shared and distinct regulation happening at the overall transcriptome level would be a logFC-logFC plot - of course, with the caveat of these two datasets originating from two different technology platforms (RNA-seq for `se_salah` and microarray for `se_paul`)

```{r}
dim(results_salah_shrink)
dim(results_paul)

results_salah_shrink$SYMBOL <- mapIds(
  org.Hs.eg.db,
  keys = rownames(results_salah_shrink),
  column = "SYMBOL",
  keytype = "ENSEMBL",
  multiVals = "first"
)

results_paul$SYMBOL <- rownames(results_paul)

merged_results <- merge.data.frame(x = as.data.frame(results_salah_shrink), 
                                   y = results_paul, 
                                   by = "SYMBOL") 
merged_results$DE_salah <- merged_results$padj <= 0.05 & !is.na(merged_results$padj)
merged_results$DE_paul <- merged_results$adj.P.Val <= 0.05

ggplot(merged_results,
       aes(x = log2FoldChange, y = logFC, col = DE_salah)) + 
  geom_point() + 
  scale_color_manual(values = c(scales::alpha("black", 0.3), "firebrick")) + 
  theme_bw() + 
  labs(x = "logFC - RNA-seq", 
       y = "logFC - microarray",
       title = "Highlighting DE genes from Salah et al.")
ggplot(merged_results,
         aes(x = log2FoldChange, y = logFC, col = DE_paul)) + 
  geom_point() + 
  scale_color_manual(values = c(scales::alpha("black", 0.3), "firebrick")) + 
  theme_bw() + 
  labs(x = "logFC - RNA-seq", 
       y = "logFC - microarray",
       title = "Highlighting DE genes from Paul et al.")
```

We can appreciate better how the gene expression regulation is to be assessed in quantitative nuances that overcome the dichotomization between significance or not, and the scatter plot of effect sizes can also better highlight the concordance in regulation direction.

One can also easily plot a geneset signature derived from one dataset (e.g. GO:0009411, response to UV, found in the microarray analysis), using the expression values of the other (Salah et al.).

```{r heatmapsig}
genes_uvresponse_paul <- res_enrich_paul_sig[res_enrich_paul_sig$ID == "GO:0009411", "geneID"] |> 
  strsplit(split = "/") |> 
  unlist()

genes_uvresponse_paul_ensembl <- mapIds(org.Hs.eg.db, 
                                        keys = genes_uvresponse_paul, 
                                        column = "ENSEMBL", 
                                        keytype = "SYMBOL")

GeneTonic::gs_heatmap(se = vsd_corrected,
                      gtl = gtl_salah_6h,
                      genelist = genes_uvresponse_paul_ensembl,
                      plot_title = "Using the DE genes from Paul et al.")

GeneTonic::gs_heatmap(se = vsd_corrected,
                      gtl = gtl_salah_6h,
                      geneset_id = "GO:0009411", 
                      plot_title = "Using the original DE genes for UV response from Salah et al.")

```

While most of the genes are shared, the higher power in DE detection from Salah et al. made it possible to identify a more comprehensive signature linked to this biological process.

Similarly to what we have just showcased, it is possible to compare other genesets and signatures across the individual datasets, e.g. apoptosis, response to DNA damage, and immune responses.

# Wrapping up

This notebook can be further expanded to include additional datasets - e.g., scenarios where some discordant regulation is expected. Having `DoReMiTra` providing a number of them in this analysis-ready format makes it easy to explore them in detail, ultimately better understanding the transcriptional changes occurring upon radiation.

Overall, data integration and comparison through well-curated radiation transcriptomic datasets available through the `DoReMiTra` collection is useful in biomarker discovery upon radiation exposure.

# Session info {-}

```{r sessioninfo}
sessionInfo()
```
