---
title: >
  Using DoReMiTra for analyzing and integrating publicly available transcriptomics radiation datasets
subtitle: >
  Supplementary file for 'DoReMiTra: An R/Bioconductor data package for orchestrating the analysis of radiation transcriptomic studies'
author:
- name: Ahmed Salah
  affiliation: 
  - Department of Radiation Oncology and Radiation Therapy, Mainz, Germany
  - Institute of Medical Biostatistics, Epidemiology and Informatics (IMBEI), Mainz, Germany
  email: ahhassan@uni-mainz.de
date: "`r format(Sys.Date(), '%d %B %Y')`"
format:
  html:
    toc: true
    number-sections: true
    embed-resources: true
    mainfont: "Atkinson Hyperlegible"
    link-external-newwindow: true
    theme: 
    - cosmo
    - ummz_theme.scss
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

# Loading the DoReMiTra Package

```{r loadlib, warning=FALSE, message=FALSE}
library("DoReMiTra")
library("mosdef")
library("org.Hs.eg.db")
library("DESeq2")
library("ggplot2")
library("ComplexHeatmap")
```

The user can use **`search_DoReMiTra_datasets()`** to filter the available DoReMiTra datasets based on metadata fields such as radiation type, organism, or experimental setting. This function helps narrow down datasets of interest before fetching them.

```{r search-datasets}
search_DoReMiTra_datasets(radiation_type = "X-ray", 
                          organism = "Homo sapiens", 
                          exp_setting = "ExVivo")
```

# Fetching datasets

After selecting the dataset of interest, **`get_DoReMiTra_data()`** can be used to fetch a `SummarizedExperiment` (SE) object directly from ExperimentHub.

```{r load-dataset}
dataset_name_1 <- "SE_Salah_2025_ExVivo"
se1 <- get_DoReMiTra_data(dataset_name_1)

## TODO: here, we can already specify we want to also check this second dataset that relates well to ours
## se2 <- get_DoReMiTra_data("")
```

# Inspecting a dataset

To quickly obtain a summary of an SE object, use **`summarize_DoReMiTra_se()`**, which provides a concise overview including the number of samples, genes, radiation types, and other metadata.

```{r summarize}
summarize_DoReMiTra_se(se1)
```

Once we retrieved the analysis-ready dataset of intereset, we can explore it and analyse it in detail.

## PCA plot

```{r PCA-plot}
# This dataset consists of 3 donors, 5 doses (0, 0.5, 1, 2, and 4Gy), and 2 timepoints (2h and 6h)

dds <- DESeqDataSet(se1, design = ~ +1)
dds$Dose <- factor(dds$Dose)
dds$Donors <- factor(dds$Donors)
dds$Dose <- relevel(dds$Dose, ref = "0Gy")
dds_t1 <- dds[, dds$Time_point == "2hrs"]
dds_t2 <- dds[, dds$Time_point == "6hrs"]

vsd_Salah <- vst(dds)
pca_plot <- plotPCA(vsd_Salah, 
                    intgroup = c("Donors", "Time_point"),
                    returnData = TRUE)

percentVar <- round(100 * attr(pca_plot, "percentVar"))

PCA<- ggplot(pca_plot, aes(x = PC1, y = PC2, color = Donors, shape = Time_point)) +
  geom_point(size = 3) +
  scale_shape_manual(values = c(16, 17)) + 
  xlab(paste0("PC1: ", percentVar[1], "% variance")) +
  ylab(paste0("PC2: ", percentVar[2], "% variance")) +
  theme_minimal()+
  theme(plot.background = element_rect(fill = "white", color = NA))

PCA

## TODO: compare to the vanilla output of...
pcaExplorer::pcaplot(vsd_Salah, intgroup = c("Donors", "Time_point"), ellipse = FALSE)
pcaExplorer::pcaplot(vsd_Salah, intgroup = "Time_point", ellipse = FALSE)
pcaExplorer::pcaplot(vsd_Salah, intgroup = "Donors", ellipse = FALSE)
pcaExplorer::pcaplot(vsd_Salah, intgroup = "Dose", ellipse = FALSE)
```

## Heatmap of the highly variable genes

```{r Heatmap}
vsd_t1 <- vst(dds_t1)

# as we have strong donor effect, we will account for it for better visualization
vsd_corrected <- vsd_t1
assay(vsd_corrected) <- limma::removeBatchEffect(assay(vsd_corrected), batch = vsd_corrected$Donors)

top_n <- 50 

rv <- rowVars(assay(vsd_corrected))
top_genes <- order(rv, decreasing = TRUE)[1:top_n]

mat <- assay(vsd_corrected)[top_genes, ]
mat_scaled <- t(scale(t(mat)))
annotation <- as.data.frame(colData(vsd_corrected)[, c("Sex","Dose"), drop = FALSE])

Heatmap(mat_scaled,
        name = "expression",
        top_annotation = HeatmapAnnotation(df = annotation))

## TODO compare this to a...
anno_genes <- data.frame(
  gene_id = rownames(dds),
  gene_name = paste0(rowData(dds)$SYMBOL, " - ", rownames(dds))
)
GeneTonic::gs_heatmap(vsd_corrected, 
                      genelist = rownames(vsd_corrected)[top_genes], 
                      anno_col_info = c("Dose", "Donors"),
                      annotation_obj = anno_genes)
```

## Using DoReMiTraExplorer

Alternatively, one can also explore  the datasets with DoReMiTraExplorer.

The command to install the app is exemplified in the following chunk

```{r exploinstall, eval = FALSE}
# Install the DoReMiTra-explorer Shiny App
devtools::install_github("AhmedSAHassan/DoReMiTraExplorer")
```

The retrieved SE objects can be explored with **`DoReMiTra_explorer()`**, an interactive Shiny application for visualization and exploratory analysis of DoReMiTra datasets. This app allows users to examine transcriptomic responses to radiation using:

-   Principal Component Analysis (PCA)
-   Boxplots for the distribution of expression values
-   Doseâ€“response gene plots
-   Heatmaps of highly variable genes

It provides flexibility for both high-level overviews and detailed inspection of radiation-induced transcriptional patterns.

To load the app, we can simply call

```{r runapp, warning=FALSE}
#Load the DoReMiTra Explorer package
library("DoReMiTraExplorer")
if(interactive()) {
  DoReMiTraExplorer::DoReMiTra_explorer(se = se1)
}
```

TODO: here, it can make sense to enter a "static screenshot" of the app to show users how it could look like. This you can generate once, and then call as a `![alttext for the image](linktoimage.png)`

# Analyzing a dataset in detail

As we see from the PCA plot, the donor and time effects are the main sources of variability. These factors should be accounted for in downstream statistical analyses.  
Specifically, we use in this case the DESeq2 statistical framework, and focus on the XX vs YY TODO comparison.

```{r analyse}
design(dds_t1)<- ~ Donors + Dose

dds_t1 <- DESeq(dds_t1)
resultsNames(dds_t1)

# we will focus on one contrast here, 1Gy vs 0Gy
results_Salah<- results(dds_t1, 
                        name = "Dose_1Gy_vs_0Gy", 
                        alpha = 0.05)
summary(results_Salah)
results_Salah_shrink <- lfcShrink(dds_t1, 
                                  res = results_Salah, 
                                  coef = "Dose_1Gy_vs_0Gy")
```

We can also visualize the up and down regulated genes via volcano plot

```{r volcanoPlot}
volcPlot <- de_volcano(
  res_de = results_Salah_shrink,
  mapping = "org.Hs.eg.db",
  labeled_genes = 50,
  logfc_cutoff = 0
)

volcPlot

# As we see here the highest X-ray sensitive genes are CDKN1A, AEN, and CD83 upon 1Gy exposure
```

TODO: We could simply run here an enrichment analysis on that result (mosdef::clupro?)

# Comparing multiple datasets with `DoReMiTra`

To compare two or more SE objects from the DoReMiTra collection, **`compare_DoReMiTra_datasets()`** summarizes key metadata information such as radiation type, dose, and time point.

```{r compare}
se2 <- get_DoReMiTra_data("SE_Paul_2013_ExVivo_GSE44201_GPL6480")

se_list<- list(
  Salah = se1, 
  Paul = se2
)
compare_DoReMiTra_datasets(se_list = se_list) |> kableExtra::kable()
```

TODO: Of course, more detailed information can be provided when integrating the two datasets at the level of differentially expressed features, or also at the pathway level

TODO: take another dataset, run DE on that, run enrichment, check some D1 genes in the D2 dataset, and viceversa. Check one regulated pathway from D1 in the D2, and viceversa

TODO: especially here, please give the datasets and the objects some meaningful names

# Session info {-}

```{r sessioninfo}
sessionInfo()
```


