---
title: "DoReMiTra Use Case"
subtitle: "Bioinformatics Group, IMBEI, University Medical Center Mainz"
author:
- name: Ahmed Salah
  affiliation: 
  - Department of Radiation Oncology and Radiation Therapy, Mainz, Germany
  - Institute of Medical Biostatistics, Epidemiology and Informatics (IMBEI), Mainz, Germany
  email: ahhassan@uni-mainz.de
date: "`r format(Sys.Date(), '%d %B %Y')`"
format:
  html:
    toc: true
    number-sections: true
    embed-resources: true
    mainfont: "Atkinson Hyperlegible"
    link-external-newwindow: true
    theme: 
    - cosmo
    - ummz_theme.scss
editor_options: 
  chunk_output_type: console
---

# Loading the DoReMiTra Package

```{r loadlib, include=FALSE}

knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)


library("DoReMiTra")
library("mosdef")
library("org.Hs.eg.db")
library("DESeq2")
library("ggplot2")
library("ComplexHeatmap")

```

The user can use **`search_DoReMiTra_datasets()`** to filter the available DoReMiTra datasets based on metadata fields such as radiation type, organism, or experimental setting. This function helps narrow down datasets of interest before fetching them.

```{r search-datasets}

search_DoReMiTra_datasets(radiation_type = "X-ray", 
                          organism = "Homo sapiens", 
                          exp_setting = "ExVivo")

```

# Fetching Datasets

After selecting the dataset of interest, **`get_DoReMiTra_data()`** can be used to fetch a `SummarizedExperiment` (SE) object directly from ExperimentHub.

```{r load-dataset}

dataset_name_1 <- "SE_Salah_2025_ExVivo"

se1 <- get_DoReMiTra_data(dataset_name_1)

```

# Summarizing a Dataset

To quickly obtain a summary of an SE object, use **`summarize_DoReMiTra_se()`**, which provides a concise overview including the number of samples, genes, radiation types, and other metadata.

```{r summarize}

summarize_DoReMiTra_se(se1)

```

# Once we retrieved the analysis-ready dataset of intereset, we can explore and analyse 

## PCA plot

```{r PCA-plot}

# This dataset consists of 3 donors, 5 doses (0, 0.5, 1, 2, and 4Gy), and 2 timepoints (2h and 6h)

dds<- DESeqDataSet(se1, design = ~ +1)

dds$Dose <- factor(dds$Dose)

dds$Donors <- factor(dds$Donors)

dds$Dose <- relevel(dds$Dose, ref = "0Gy")

dds_t1 <- dds[, dds$Time_point == "2hrs"]

dds_t2 <- dds[, dds$Time_point == "6hrs"]


vsd_Salah <- vst(dds)

pca_plot <- plotPCA(vsd_Salah, 
                    intgroup = c("Donors", "Time_point"),
                    returnData = TRUE)

percentVar <- round(100 * attr(pca_plot, "percentVar"))

PCA<- ggplot(pca_plot, aes(x = PC1, y = PC2, color = Donors, shape = Time_point)) +
  geom_point(size = 3) +
  scale_shape_manual(values = c(16, 17)) + 
  xlab(paste0("PC1: ", percentVar[1], "% variance")) +
  ylab(paste0("PC2: ", percentVar[2], "% variance")) +
  theme_minimal()+
  theme(plot.background = element_rect(fill = "white", color = NA))

PCA

```

## Heatmap of the highly variable genes

```{r Heatmap}
vsd_t1 <- vst(dds_t1)

# as we have strong donor effect, we will account for it for better visualization

vsd_corrected <- vsd_t1
assay(vsd_corrected) <- limma::removeBatchEffect(assay(vsd_corrected), batch = vsd_corrected$Donors)

top_n <- 50 

rv <- rowVars(assay(vsd_corrected))
top_genes <- order(rv, decreasing = TRUE)[1:top_n]

mat <- assay(vsd_corrected)[top_genes, ]
mat_scaled <- t(scale(t(mat)))
annotation <- as.data.frame(colData(vsd_corrected)[, c("Sex","Dose"), drop = FALSE])


Heatmap(mat_scaled,
        name = "expression",
        top_annotation = HeatmapAnnotation(df = annotation))

```

## Alternatively, one can also explore  the datasets with DoReMiTraExplorer

The retrieved SE objects can be explored with **`DoReMiTra_explorer()`**, an interactive Shiny application for visualization and exploratory analysis of DoReMiTra datasets. This app allows users to examine transcriptomic responses to radiation using:

-   Principal component analysis (PCA)

-   Boxplots

-   Doseâ€“response gene plots

-   Heatmaps of highly variable genes

It provides flexibility for both high-level overviews and detailed inspection of radiation-induced transcriptional patterns.

### Installing and loading the app

```{r Shiny-load}

# Install the DoReMiTra-explorer Shiny App

devtools::install_github("AhmedSAHassan/DoReMiTraExplorer")

#Load the DoReMiTra Explorer package

suppressWarnings(library("DoReMiTraExplorer"))

```

```{r run-app, eval=FALSE}

DoReMiTra_explorer(se1)
```

## As we see from the PCA plot, the donor and time effects are the main sources of variability. These factors should be accounted for in downstream statistical analyses.

```{r analyse}

design(dds_t1)<- ~ Donors + Dose

dds_t1 <- DESeq(dds_t1)

resultsNames(dds_t1)

# we will focus on one contrast here, 1Gy vs 0Gy

resuls_Salah<- results(dds_t1, 
                       name = "Dose_1Gy_vs_0Gy", 
                       alpha = 0.05)

summary(resuls_Salah)

resuls_Salah_shrink <- lfcShrink(dds_t1, 
                                 res = resuls_Salah, 
                                 coef = "Dose_1Gy_vs_0Gy")
```

## We can also visualize the up and down regulated genes via volcano plot
```{r volcanoPlot}

volcPlot <- de_volcano(
  res_de = resuls_Salah_shrink,
  mapping = "org.Hs.eg.db",
  labeled_genes = 50,
  logfc_cutoff = 0
)

volcPlot

# As we see here the highest X-ray sensitive genes are CDKN1A, AEN, and CD83 upon 1Gy exposure
```

# Comparing Multiple Datasets

To compare two or more SE objects from the DoReMiTra collection, **`compare_DoReMiTra_datasets()`** summarizes key metadata information such as radiation type, dose, and time point.

```{r compare}

se2 <- get_DoReMiTra_data("SE_Paul_2013_ExVivo_GSE44201_GPL6480")

se_list<- list(
  Salah = se1, 
  Paul = se2
)
compare_DoReMiTra_datasets(se_list = se_list) |> kableExtra::kable()
```
